api/billingd/service/service.go

```diff
@@ -455,8 +460,23 @@ func (s *Service) createCustomer(
 	lck.Acquire()
 	defer lck.Release()
 
-	doc := &Customer{
+	doc, err := s.getCustomer(ctx, "_id", params.Key)
+	if err != nil && !errors.Is(err, mongo.ErrNoDocuments) {
+		return nil, err
+	} else if err == nil {
+		return nil, ErrCustomerExists
+	}
+
+	customer, err := s.stripe.Customers.New(&stripe.CustomerParams{
+		Email: stripe.String(params.Email),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	doc = &Customer{
 		Key:         params.Key,
+		CustomerID:  customer.ID,
```

Comment:
Since this method now takes the per-customer lock, we can safely use `getCustomer` to determine if the customer already exists, vs. relying on a write operation (which needed to come before the Stripe customer creation to avoid creating a bunch of unused Stripe customers in the event the doc already exists).

Reply:
The result is avoiding this while problem of the empty `customer_id` string.